argilla/docs/getting_started/huggingface-spaces.md

```diff
@@ -0,0 +1,147 @@
+
+    | Secret name     | Type | Description | Default value   |
+    |-------------------------------|----------|-------------|----------------|
+    | `OWNER_USERNAME` | string | A username to log in with owner permissions | owner |
+    | `OWNER_PASSWORD` | string | A password for the `OWNER_USERNAME` | 12345678 |
+    | `OWNER_API_KEY` | string or online generator |An `api_key` to interact with the SDK with owner permissions | owner.apikey |
+    | `ADMIN_USERNAME` | string |A username to log in with admin permissions | admin |
+    | `ADMIN_PASSWORD` | string | A password for the `ADMIN_USERNAME` | 12345678 |
+    | `ADMIN_API_KEY` | string or online generator|An `api_key` to interact with the SDK with admin permissions | admin.apikey |
+    | `ANNOTATOR_USERNAME` | string |A username to log in with annotator permissions | argilla |
+    | `ANNOTATOR_PASSWORD` | string |A password for the `ANNOTATOR_PASSWORD` | 12345678 |
+    | `ARGILLA_WORKSPACE` | string | The name of the default workspace | admin |
+    | `OAUTH2_HUGGINGFACE_CLIENT_ID` | ID | The Client ID of your connected app | |
+    | `OAUTH2_HUGGINGFACE_CLIENT_SECRET` | ID | The app secret of your connected app | |
+
+
+## Sign in with Hugging Face
+
+The Hugging Face authentication allows you to give access to your Argilla Space to users that are logged in to the Hugging Face Hub.
+
+!!! tip
+    This feature is especially helpful for public crowdsourcing projects.
+
+    To have more control over who can log in to the Space, you can set this up in a private Space so that only members of your organization can sign in.
+
+    Alternatively, you can [create users](../how_to_guides/user.md) and access with their credentials.
+
+!!! note
+    If you duplicated an Argilla Space with OAuth instead of creating one from the template, make sure to also follow the next steps.
+
+!!! warning
+    Make sure to enable the Hugging Face authentication before adding any data, as the Space will need to be restarted with a "Factory build", removing all the previous data.
+
+1. [Create an OAuth App in Hugging Face](https://huggingface.co/settings/applications/new) and fill in the information:
+
+    | Field        | Value |
+    |--------------|----------|
+    | Homepage URL | [`[Your Argilla Space direct URL]`](#deploy-argilla-on-spaces) |
+    | Logo URL | `[Your Argilla Space direct URL]/favicon.ico` |
+    | Scopes | `openid` and `profile` |
+    | Redirect URL | `[Your Argilla Space Direct URL]/oauth/huggingface/callback`
+
+2. Add the created Client ID and App Secret to your Space. It can be done in two different ways.
+
+    === "As secrets"
+
+        Add them to your Space in "Settings" > "New secret".
+
+        | Secret name         | Value |
+          |-------------------------------|----------|
+          | `OAUTH2_HUGGINGFACE_CLIENT_ID` | [Your Client ID] |
+          | `OAUTH2_HUGGINGFACE_CLIENT_SECRET` | [Your App Secret] |
+
+    === "As environment variables in `.oauth.yaml`"
```

Comment:
Should we maybe not add this option in the docs so that we encourage people to use this as secrets always for better security?

Reply:
I agree that I prefer secrets, but I would leave it until we can add some general basic guidelines about how to enable OAuth not only with HF but with other platforms (e.g. Slack or Github, as some users have been checking)